library/components/TFiltersSection.vue

```diff
@@ -172,16 +179,27 @@ export default {
       }
       return urlFilters;
     },
-    removeFilter(urlFilters) {
-      for (let filter of urlFilters) {
-        this.deleteFilter({ name: filter });
-        if (
-          Object.prototype.hasOwnProperty.call(this.selectedFilters, filter)
-        ) {
-          this.$delete(this.selectedFilters, filter);
+    removeFilter(urlFilters, vfname) {
+      for (let filterName of urlFilters) {
+        const filterIndex = this.selectedFilters.findIndex(
+          (f) => f.name === filterName
+        );
+        if (filterIndex > -1) {
+          this.selectedFilters.splice(filterIndex, 1);
         }
+        this.deleteFilter({ name: filterName });
+        this.filters = this.filters.filter((sf) => sf.name !== filterName);
```

Comment:
This shouldn't be done or needed. `this.filters` is an inherited state from the Vuex stores and the `deleteFilter` called from the line above should take care of doing exactly what this line is trying to do.

Reply:
Yep, removed!